### ISSUE TITLE
[Serializer] Error when using a named argument in the "normalize" method of the NormalizerInterface.

### ISSUE THREAD

**Issue by AlessandroAmos** [symfony/symfony#53326]
### Symfony version(s) affected

6.0 - 7.0

### Description

The implementation of the NormalizerInterface in the Serializer class uses a different name for the first argument ($data) compared to the interface ($object), causing an error when attempting to invoke the normalize method with named arguments.

### How to reproduce

Simply make a call to normalize by passing named arguments.
```php
public function userToArray(NormalizerInterface $normalizer, User $user): array
{
    return $normalizer->normalize(
        object: $user,
        format: 'json',
        context: [
            'user:hide_email' => true,
        ]
    );
}
```

I have an exception "Unknown named parameter $object."

### Possible Solution

Rename the argument name $object to $data in NormalizerInterface -> normalize

**Comment by xabbuh** [comment]
I would not consider this a bug as our BC promise does not cover argument names (see https://symfony.com/doc/current/contributing/code/bc.html#note-10).

**Comment by nesl247** [comment]
> I would not consider this a bug as our BC promise does not cover argument names (see [symfony.com/doc/current/contributing/code/bc.html#note-10](https://symfony.com/doc/current/contributing/code/bc.html#note-10)).

I wouldn't say it's a BC issue though, and thus would have nothing to do with the BC promise. I don't see it as a BC issue because we're not talking about compatibility between versions.

It's a mismatch in names which makes named arguments unusable in this instance.

**Comment by nesl247** [comment]
Any chance this could be revisited?

**Comment by xabbuh** [comment]
This has been addressed in #54880 and will ship with 7.2.

**Comment by nesl247** [comment]
I was finally able to test this as we're working on upgrading to Symfony 7.2. I would not consider this completed. You cannot inject `NormalizerInterface` and used named arguments. If `ObjectNormalizer` is used (supports returns true), then named arguments will fail because it uses `object` not `data` per the interface.
